holopy/scattering/tests/test_calculations.py

```diff
@@ -30,65 +31,128 @@
 from holopy.core import detector_grid
 from holopy.core.tests.common import assert_obj_close
 from holopy.scattering.calculations import *
+from holopy.scattering.errors import MissingParameter
 
-scatterer = Sphere(n=1.6, r=.5, center=(5, 5, 5))
-medium_index = 1.33
-locations = detector_grid(shape=(20, 20), spacing=.1)
-wavelen = 0.66
-polarization=(0, 1)
+import xarray as xr
 
-@attr("fast")
-def test_calc_holo():
-    holo = calc_holo(locations, scatterer, medium_index, wavelen, polarization)
+SCATTERER = Sphere(n=1.6, r=.5, center=(5, 5, 5))
+MED_INDEX = 1.33
+LOCATIONS = detector_grid(shape=(20, 20), spacing=.1)
+WAVELEN = 0.66
+POL = (0, 1)
 
 
-def test_calc_field():
-    field = calc_field(locations, scatterer, medium_index, wavelen, polarization)
+class TestCalculations(unittest.TestCase):
+    @attr("fast")
+    def test_calc_holo(self):
+        holo = calc_holo(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
+    def test_calc_field(self):
+        field = calc_field(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
-@attr("fast")
-def test_calc_cross_section():
-    cross = calc_cross_sections(scatterer, medium_index, wavelen, polarization)
+    @attr("fast")
+    def test_calc_cross_sections(self):
+        cross = calc_cross_sections(SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
+    def test_calc_intensity(self):
+        intensity = calc_intensity(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
-def test_calc_intensity():
-    intensity = calc_intensity(locations, scatterer, medium_index, wavelen, polarization)
+    @attr("fast")
+    def test_calc_scat_matrix(self):
+        matr = calc_scat_matrix(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN)
+        self.assertTrue(True)
 
+    def test_finalize(self):
+        detector = finalize(LOCATIONS.values, LOCATIONS)
+        self.assertTrue(True)
 
-@attr("fast")
-def test_calc_scat_matrix():
-    matr = calc_scat_matrix(locations, scatterer, medium_index, wavelen)
+    def test_scattered_field_to_hologram(self):
+        size = 3
+        coords = np.linspace(0, 1, size)
+        scat = xr.DataArray(np.array([0.5, 0, 0]), coords=[('vector', coords)])
+        ref = xr.DataArray(np.array([0.5, 0, 0]), coords=[('vector', coords)])
+        normals = np.array((0, 0, 1))
+        holo = scattered_field_to_hologram(scat, ref, normals)
+        self.assertEqual(holo.values.mean(), 1.)
```

Comment:
It might be a little clearer to calculate the correct answer here, e.g.
```python
correct_holo = (np.abs(scat + ref)**2).sum(dim='vector')
self.assertEqual(holo.values[0], correct_holo.values[0])
```
or something. As is it's a little difficult to see why 1.0 is the correct answer. 

That being said, I think I have to redo this function anyway according to #283 so this might not be worth your time.

Reply:
resolved by 20c0e44